public/app/plugins/datasource/cloudwatch/components/MetricsQueryFieldsEditor.tsx

```diff
@@ -120,12 +120,25 @@ export function MetricsQueryFieldsEditor({
             />
           </QueryInlineField>
 
-          <QueryInlineField label="Stats">
-            <Stats
-              stats={datasource.standardStatistics.map(toOption)}
-              values={metricsQuery.statistics}
-              onChange={(statistics) => onQueryChange({ ...metricsQuery, statistics })}
-              variableOptionGroup={variableOptionGroup}
+          <QueryInlineField label="Statistic">
+            <Segment
+              allowCustomValue
+              value={query.statistic}
+              options={[
+                ...datasource.standardStatistics.filter((s) => s !== query.statistic).map(toOption),
+                variableOptionGroup,
+              ]}
+              onChange={({ value: statistic }) => {
+                if (
+                  !datasource.standardStatistics.includes(statistic) &&
+                  !/^p\d{2}(?:\.\d{1,2})?$/.test(statistic) &&
+                  !statistic.startsWith('$')
```

Comment:
this may be hiding an error if the statistic is not properly formatted?

Reply:
should be fine! unfortunately, unit testing the segment component is very painful. however we plan to rewrite the cloudwatch query editor to use react select in 8.2, so I'll add proper unit tests for this then.